test/CSharp/CodeCracker.Test/Verifiers/CodeFixVerifier.cs

```diff
@@ -42,14 +42,15 @@ protected virtual CodeFixProvider GetBasicCodeFixProvider()
         /// <param name="newSource">A class in the form of a string after the CodeFix was applied to it</param>
         /// <param name="codeFixIndex">Index determining which codefix to apply if there are multiple</param>
         /// <param name="allowNewCompilerDiagnostics">A bool controlling whether or not the test will fail if the CodeFix introduces other warnings after being applied</param>
-        protected async Task VerifyCSharpFixAsync(string oldSource, string newSource, int? codeFixIndex = null, bool allowNewCompilerDiagnostics = false, bool formatBeforeCompare = true)
+        /// <param name="codeFixProvider">The codefix to be applied to the code wherever the relevant Diagnostic is found</param>
+        protected async Task VerifyCSharpFixAsync(string oldSource, string newSource, int? codeFixIndex = null, bool allowNewCompilerDiagnostics = false, bool formatBeforeCompare = true, CodeFixProvider codeFixProvider = null)
```

Comment:
The added codeFixProvider parameter made possible to pass a mock object to VerifyCSharpFix Async method.
I couldn't override the CSharpCodeProvider method, because the multithread nature of our tests.